app/controllers/users_controller.rb

```diff
@@ -0,0 +1,13 @@
+class Users < Devise::OmniauthCallbacksController
+  class OmniauthCallbacksController
+    def google_oauth2
+      @user = User.from_omniauth(request.env['omniauth.auth'])
+
+      if @user.persisted?
```

Comment:
```
return sign_in_and_redirect @user, event: :authentication if @user.persisted?
redirect_to new_user_registration_url, alert: @user.errors.full_messages.join("\n")
```
unless it's more than 100 characters. Ruby style guidelines usually tell you to do clauses like that instead of traditional if/then/else structures (it makes the code way cleaner and forces you to delegate correctly when the code is more complex).

I know they are really long and we don't follow them 100% but you can find some good ideas on [ruby](https://github.com/bbatsov/ruby-style-guide) and [rails](https://github.com/bbatsov/rails-style-guide) style guides.